pkg/nakadi/nakadi.go

```diff
@@ -0,0 +1,125 @@
+package nakadi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// Nakadi defines an interface for talking to the Nakadi API.
+type Nakadi interface {
+	ConsumerLagSeconds(ctx context.Context, subscriptionID string) (int, error)
+	UnconsumedEvents(ctx context.Context, subscriptionID string) (int, error)
+}
+
+// Client defines client for interfacing with the Nakadi API.
+type Client struct {
+	nakadiEndpoint string
+	http           *http.Client
+}
+
+// NewNakadiClient initializes a new Nakadi Client.
+func NewNakadiClient(nakadiEndpoint string, client *http.Client) *Client {
+	return &Client{
+		nakadiEndpoint: nakadiEndpoint,
+		http:           client,
+	}
+}
+
+func (c *Client) ConsumerLagSeconds(ctx context.Context, subscriptionID string) (int, error) {
+	stats, err := c.stats(ctx, subscriptionID)
+	if err != nil {
+		return 0, err
+	}
+
+	var maxConsumerLagSeconds int
+	for _, partition := range stats.Partitions {
+		if partition.ConsumerLagSeconds > maxConsumerLagSeconds {
+			maxConsumerLagSeconds = partition.ConsumerLagSeconds
+		}
+	}
+
+	return maxConsumerLagSeconds, nil
+}
+
+func (c *Client) UnconsumedEvents(ctx context.Context, subscriptionID string) (int, error) {
+	stats, err := c.stats(ctx, subscriptionID)
+	if err != nil {
+		return 0, err
+	}
+
+	var unconsumedEvents int
+	for _, partition := range stats.Partitions {
+		unconsumedEvents += partition.UnconsumedEvents
+	}
+
+	return unconsumedEvents, nil
+}
+
+type statsResp struct {
+	Items []statsSubscription `json:"items"`
+}
+
+type statsSubscription struct {
```

Comment:
I think this should be statsEventType https://nakadi.io/manual.html#definition_SubscriptionEventTypeStats

Reply:
Maybe add unused `event_type` field for clarity.

Reply:
Changed!